### ISSUE TITLE
Changing iconset with missing icon causes error

### ISSUE THREAD

**Issue by potmo** [richardszalay/fastlane-plugin-act#3]
I'm getting an error `no implicit conversion of nil into String` from the `act` plugin when changing the iconset of  an ipa.

It turns out that https://github.com/richardszalay/fastlane-plugin-act/blob/master/lib/fastlane/plugin/act/helper/icon_patcher.rb#L47 is line that crashes.
There is a missing `filename` entry in the `Contents.json` of my `AppIcon.appiconset`

When opening the iconset in xCode it seems to be fine.

![screen shot 2016-10-21 at 11 17 03](https://cloud.githubusercontent.com/assets/1089422/19593238/e9c0e1a6-977f-11e6-9526-4d6401061837.png)

`Assets.xcassets` -> `AppIcon.appiconset` -> `Contents.json` is 

``` json
{
  "images" : [
    {
      "idiom" : "iphone",
      "size" : "20x20",
      "scale" : "2x"
    },
    {
      "idiom" : "iphone",
      "size" : "20x20",
      "scale" : "3x"
    },
    {
      "size" : "29x29",
      "idiom" : "iphone",
      "filename" : "AppIcon29x29@2x.png",
      "scale" : "2x"
    },
    {
      "size" : "29x29",
      "idiom" : "iphone",
      "filename" : "AppIcon29x29@3x.png",
      "scale" : "3x"
    },
    {
      "size" : "40x40",
      "idiom" : "iphone",
      "filename" : "AppIcon40x40@2x.png",
      "scale" : "2x"
    },
    {
      "size" : "40x40",
      "idiom" : "iphone",
      "filename" : "AppIcon40x40@3x.png",
      "scale" : "3x"
    },
    {
      "size" : "60x60",
      "idiom" : "iphone",
      "filename" : "AppIcon60x60@2x.png",
      "scale" : "2x"
    },
    {
      "size" : "60x60",
      "idiom" : "iphone",
      "filename" : "AppIcon60x60@3x.png",
      "scale" : "3x"
    }
  ],
  "info" : {
    "version" : 1,
    "author" : "xcode"
  }
}
```

The first two elements does not contain a file. It seems strange but it is setup in xCode so I guess the data is valid according to xCode anyway.

``` json
 {
   "idiom" : "iphone",
   "size" : "20x20",
   "scale" : "2x"
 }
```

Preferred would be to either ignore elements without a `filename` or showing a better error would be nice.

I'm using xCode `Version 8.0 (8A218a)` with ios 10.0 as target.

**Comment by potmo** [comment]
Manually removing the element without `filename` fixes the crash but opening the iconset in xCode puts the element back.

**Comment by richardszalay** [comment]
Interesting, thanks for letting me know.

I'm pretty flat out at the moment, but I'd be happy to accept a PR if you're keen. If you do, it would be great if you could add a unit test for it.

**Comment by richardszalay** [comment]
Out of interest, what's xcode's behaviour with that type of icon? Does it exclude it to does it resize one of the other icons?

**Comment by potmo** [comment]
I'm not sure. It's for notifications I guess and I don't have that set up yet.

Maybe I'll have some time next week to create a fix for this.

Do you think dropping the element is better than throwing a more informative error?
I guess that depends on what xCode will do. If xCode will resize one of the existing icons then just a warning would be better.

**Comment by richardszalay** [comment]
It shouldn't be to difficult to find out what code does. Assign it as your app's icon, and then build and export an IPA. Then just open the IPA with archiver and take a look at which icons it added to the AppName.app/ folder.

**Comment by richardszalay** [comment]
I ended up running into this myself, so I've released 1.0.1 of act that includes a fix (filename-less images are skipped)
